websites/api.py

```diff
@@ -0,0 +1,121 @@
+
+import boto3
+import yaml
+from django.conf import settings
+
+from main.utils import get_s3_object_and_read
+from websites.constants import CONTENT_TYPE_PAGE, CONTENT_TYPE_FILE, COURSE_HOME
+from websites.models import Website, WebsiteContent
+
+log = logging.getLogger(__name__)
+
+
+def import_ocw2hugo_content(bucket, prefix, website):
+    """
+    Import all content files for an ocw course from hugo2ocw output
+
+    Args:
+        bucket (s3.Bucket): S3 bucket
+        prefix (str): S3 prefix for filtering by course
+        website (Website): Website to import content for
+    """
+    for resp in bucket.meta.client.get_paginator("list_objects").paginate(
+        Bucket=bucket.name, Prefix=f"{prefix}content/courses/{website.url_path}"
+    ):
+        for obj in resp["Contents"]:
+            s3_content = get_s3_object_and_read(bucket.Object(obj["Key"])).decode()
+            s3_content_parts = [
+                part
+                for part in re.split(re.compile(r"^---\n", re.MULTILINE), s3_content)
+                if part
+            ]
+            parent = None
+            if len(s3_content_parts) >= 1:
+                content_json = yaml.load(s3_content_parts[0], Loader=yaml.Loader)
+                menu = content_json.get("menu", None)
+                if menu:
+                    # This is a page
+                    menu_values = list(menu.values())[0]
+                    uuid = menu_values.get("identifier")
+                    if uuid == COURSE_HOME:
+                        uuid = website.uuid
+                    parent_uuid = menu_values.get("parent", None)
+                    content_type = CONTENT_TYPE_PAGE
+                else:
+                    # This is a file
+                    uuid = content_json.get("uid")
+                    parent_uuid = content_json.get("parent", None)
+                    content_type = CONTENT_TYPE_FILE
+                if parent_uuid:
+                    parent, _ = WebsiteContent.objects.get_or_create(
+                        website=website, uuid=parent_uuid
+                    )
+                if not uuid:
+                    # This code block is a temporary hack until every hugo2ocw output file has a uuid
+                    WebsiteContent.objects.update_or_create(
+                        website=website,
+                        hugo_filepath=obj["Key"].replace(prefix, ""),
+                        defaults={
+                            "metadata": content_json,
+                            "markdown": (
+                                s3_content_parts[1]
+                                if len(s3_content_parts) == 2
+                                else None
+                            ),
+                            "uuid": uuid4(),
+                            "parent": parent,
+                            "title": content_json.get("title"),
+                            "type": content_type,
+                        },
+                    )
```

Comment:
Define `defaults` once as a var and use it here and in the `else`

Reply:
The defaults are slightly different in each code block (uuid, hugo_filepath), and once the hack becomes unnecessary there will be only usage, but I'll create a `base_defaults` variable with the common attributes and update as necessary for each block.

Reply:
ok, yeah, i missed that